### ISSUE TITLE
Hart85.assign_power_from_states should return a 1D array?

### ISSUE THREAD

**Issue by JackKelly** [nilmtk/nilmtk#435]
After fixing #434 (and running the minimal code example I copied into that issue queue), we now get a new error:

``` python
---------------------------------------------------------------------------
Exception                                 Traceback (most recent call last)
<ipython-input-4-c0024030b974> in <module>()
----> 1 __pyfile = open('''/tmp/py29706Qfe''');exec(compile(__pyfile.read(), '''/home/dk3810/temp/nilmtk_issue434.py''', 'exec'));__pyfile.close(); import os; os.remove('''/tmp/py29706Qfe''')

/home/dk3810/temp/nilmtk_issue434.py in <module>()
     28     output,
     29     sample_period=chooseSample)
---> 30 output.close()

/home/dk3810/workspace/python/nilmtk/nilmtk/disaggregate/hart_85.pyc in disaggregate(self, mains, output_datastore, **load_kwargs)
    422             measurement = chunk.name
    423             power_df = self.disaggregate_chunk(
--> 424                 chunk, prev, transients)
    425 
    426             cols = pd.MultiIndex.from_tuples([chunk.name])

/home/dk3810/workspace/python/nilmtk/nilmtk/disaggregate/hart_85.pyc in disaggregate_chunk(self, chunk, prev, transients)
    312         prev = states.iloc[-1].to_dict()
    313         power_chunk_dict = self.assign_power_from_states(states, prev)
--> 314         return pd.DataFrame(power_chunk_dict, index=chunk.index)
    315 
    316     def assign_power_from_states(self, states_chunk, prev):

/usr/local/lib/python2.7/dist-packages/pandas/core/frame.pyc in __init__(self, data, index, columns, dtype, copy)
    212                                  dtype=dtype, copy=copy)
    213         elif isinstance(data, dict):
--> 214             mgr = self._init_dict(data, index, columns, dtype=dtype)
    215         elif isinstance(data, ma.MaskedArray):
    216             import numpy.ma.mrecords as mrecords

/usr/local/lib/python2.7/dist-packages/pandas/core/frame.pyc in _init_dict(self, data, index, columns, dtype)
    339 
    340         return _arrays_to_mgr(arrays, data_names, index, columns,
--> 341                               dtype=dtype)
    342 
    343     def _init_ndarray(self, values, index, columns, dtype=None,

/usr/local/lib/python2.7/dist-packages/pandas/core/frame.pyc in _arrays_to_mgr(arrays, arr_names, index, columns, dtype)
   4801 
   4802     # don't force copy because getting jammed in an ndarray anyway
-> 4803     arrays = _homogenize(arrays, index, dtype)
   4804 
   4805     # from BlockManager perspective

/usr/local/lib/python2.7/dist-packages/pandas/core/frame.pyc in _homogenize(data, index, dtype)
   5110                 v = lib.fast_multiget(v, oindex.values, default=NA)
   5111             v = _sanitize_array(v, index, dtype=dtype, copy=False,
-> 5112                                 raise_cast_failure=False)
   5113 
   5114         homogenized.append(v)

/usr/local/lib/python2.7/dist-packages/pandas/core/series.pyc in _sanitize_array(data, index, dtype, copy, raise_cast_failure)
   2718     elif subarr.ndim > 1:
   2719         if isinstance(data, np.ndarray):
-> 2720             raise Exception('Data must be 1-dimensional')
   2721         else:
   2722             subarr = _asarray_tuplesafe(data, dtype=dtype)

Exception: Data must be 1-dimensional
```

the bug here is because, on line 314 of `hart_85.py`, `pd.DataFrame(power_chunk_dict, index=chunk.index)`, the DataFrame constructor is expecting a dict of 1D arrays.  But it's getting a dict of 2D arrays:

``` python

ipdb> power_chunk_dict
{0: array([[0, 0],
       [0, 0],
       [0, 0],
       ..., 
       [0, 0],
       [0, 0],
       [0, 0]]), 1: array([[0, 0],
       [0, 0],
       [0, 0],
       ..., 
       [0, 0],
       [0, 0],
       [0, 0]]), 2: array([[0, 0],
       [0, 0],
       [0, 0],
       ..., 
       [0, 0],
       [0, 0],
       [0, 0]]), 3: array([[0, 0],
       [0, 0],
       [0, 0],
       ..., 
       [0, 0],
       [0, 0],
       [0, 0]])}
```

But I'm not sure what the appropriate mathematical operator is to change this into a 1D array.  Any thoughts, @nipunreddevil?

**Comment by gjwo** [comment]
I can reproduce this error on my data, so I am interested in the fix.

**Comment by JackKelly** [comment]
I think we need @nipunreddevil's help (but he's crazy busy, I believe).

**Comment by gjwo** [comment]
@JackKelly I'm no python expert so I don't know how to do this (and I may be completely wrong), but I think the issue here is not how to transform it into a 1-dimensional data but split it into two 1-dimensional results one for the Active, and one for the Reactive and assign these separately to the relevant columns after the return to the calling location. perhaps in a similar manner to the way find_steady_states_transients handles the results from find_steady_states

**Comment by nipunreddevil** [comment]
@gjwo: I'll get around this one on Monday.

**Comment by rajatgupta1234** [comment]
@gjwo 
The issue has been solved.
`h.train(mains,columns=[('power','active'),('power','reactive')])` works fine without any error.
`output = HDFDataStore('output.h5', 'w')`
`df=h.disaggregate(mains,output)` also works fine now. Moreover for avoiding any confusion, Hart's documentation will be added soon.
The issue has been solved using #794
